src/main/kotlin/io/github/anki/anki/service/CardsService.kt

```diff
@@ -1,32 +1,74 @@
 package io.github.anki.anki.service
 
+import io.github.anki.anki.controller.exceptions.CardDoesNotExistException
 import io.github.anki.anki.repository.mongodb.CardRepository
+import io.github.anki.anki.repository.mongodb.document.MongoCard
 import io.github.anki.anki.service.model.Card
 import io.github.anki.anki.service.model.mapper.toCard
 import io.github.anki.anki.service.model.mapper.toMongo
+import org.bson.types.ObjectId
 import org.slf4j.LoggerFactory
 import org.springframework.stereotype.Service
 
 @Service
 class CardsService(
     private val cardRepository: CardRepository,
+    private val deckService: DeckService,
 ) {
-
-    fun createNewCard(card: Card): Card {
+    fun createNewCard(userId: String, card: Card): Card {
+        deckService.getDeckByIdAndUserId(card.deckId, userId)
         LOG.info("Creating new card: {}", card)
-        return cardRepository.insert(
-            card.toMongo(),
-        )
+        return cardRepository.insert(card.toMongo())
             .toCard()
             .also { LOG.info("Successfully saved new card: {}", it) }
     }
 
-    fun deleteCard(cardId: String) {
+    fun getAllCardsFromDeck(deckId: String, userId: String): List<Card> {
+        deckService.getDeckByIdAndUserId(deckId, userId)
+        LOG.info("Getting MongoCards by deckId = {}", deckId)
+        return cardRepository.findByDeckId(ObjectId(deckId))
+            .map { it.toCard() }
+            .also { LOG.info("Got MongoCards from DB {}", it) }
+    }
+
+    fun updateCard(userId: String, card: Card): Card {
+        deckService.getDeckByIdAndUserId(card.deckId, userId)
+        val mongoCard = getCardById(card.id!!)
+        val updatedMongoCard = mongoCard.update(card)
+        if (mongoCard == updatedMongoCard) {
+            LOG.info("No changes. Nothing to update")
```

Comment:
1. Imagine that you have a thousands of these log messages, then you how determinate context of this message? The any log message should contain context information to help determine chain from user request to database.
2. Is you really need this log? Can you imagine situation when this log can be helpful?

Reply:
ok i will remove it